SwiftMongoDB/MongoCollection.swift

```diff
@@ -55,24 +55,39 @@ public class MongoCollection {
         try error.throwIfError()
     }
     
-    public func find(query: DocumentData = DocumentData(), flags: QueryFlags = QueryFlags.None, skip: Int = 0, limit: Int = 0, batchSize: Int = 0) throws -> [MongoDocument] {
+    public func find(query: DocumentData = DocumentData(), fields: DocumentData? = nil, flags: QueryFlags = QueryFlags.None, skip: Int = 0, limit: Int = 0, batchSize: Int = 0) throws -> [MongoDocument] {
 
         var query = try MongoBSON(data: query).bson
-
         // standard options - should be customizable later on
-        let cursor = MongoCursor(
-            collection: self,
-            operation: .Find,
-            query: &query,
-            options: (
-                queryFlags: flags.rawFlag,
-                skip: skip,
-                limit: limit,
-                batchSize: batchSize
-            )
+        let options = (
+            queryFlags: flags.rawFlag,
+            skip: skip,
+            limit: limit,
+            batchSize: batchSize
         )
-
-        let documents = try cursor.getDocuments()
+        
+        var cursor: MongoCursor
```

Comment:
You can use `let` instead of `var` here (the compiler knows it _will_ be initialized), but I think there's a better way to do this chunk anyway. Something like this, maybe?

```swift
let fields: MongoBSON? = fields != nil ? try MongoBSON(data: fields!) : fields
let cursor = MongoCursor(
cursor = MongoCursor(
    collection: self,
    operation: .Find,
    query: &query,
    fields: &fields?.bson,
    options: options
)
```

Or something like that.